aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -58,23 +69,202 @@
+                                    software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                            .channelUri("channelUri2")
+                                            .eventPublishers(EventPublisher.ANOMALY_DETECTION.toString())
+                                            .build()
+                            ))
+                            .build())
+                    .build());
+
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(removeNotificationChannelRequest), any());
+            verifyNoMoreInteractions(proxy);
+
+        }
+
+        @Test
+        public void itDoesNotAddOrDeleteNotificationChannelWithNoChangeSet() {
+            doReturn(GetNotificationConfigurationResponse.builder()
+                    .notificationConfiguration(NotificationConfiguration.builder()
+                            .channels(Channel.builder()
+                                            .id("channelId")
+                                            .uri("channelUri")
+                                            .eventPublishers(EventPublisher.ANOMALY_DETECTION)
+                                            .build(),
+                                    Channel.builder()
+                                            .id("channelId2")
+                                            .uri("channelUri2")
+                                            .build())
+                            .build())
+                    .build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).anomalyDetectionNotificationConfiguration(
+                    AnomalyDetectionNotificationConfiguration.builder()
+                            .channels(ImmutableList.of(
+                                    software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                            .channelUri("channelUri")
+                                            .eventPublishers(EventPublisher.ANOMALY_DETECTION.toString())
+                                            .build(),
+                                    software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                            .channelUri("channelUri2")
+                                            .eventPublishers(EventPublisher.ANOMALY_DETECTION.toString())
+                                            .build()
+                                    ))
+                            .build())
+                    .build());
+
```

Comment:
Since this behavior is non-trivial, consider extracting it into a separate class, and then writing tests for that class. That would simplify the tests for the update handler.